components/index/LocationSearchBar.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback } from "react";
+import GoogleSearchBar from "./GoogleSearchBar";
+import DebouncedInput from "../common/DebouncedInput";
+import { USDatasetSource } from "@/components/index/SearchButton";
+
+type Props = {
+  placeholder?: string;
+  className?: string;
+  onChange: (value: string) => void;
+  value?: string;
+  dsSource: USDatasetSource;
+};
+
+const LocationSearchBar: React.FC<Props> = ({
+  placeholder,
+  value,
+  onChange,
+  className,
+  dsSource,
+}: Props) => {
+  const renderSearchBar = useCallback(() => {
+    switch (dsSource) {
```

Comment:
this works, but what I would've done is decouple the logic of computing location value, then use a switch case on that. so the UI for each location bar will stay the same but has different logic for each different dsSource.

As an example:
```
locationCompute(value: string){
  return value.lower()
}
googleLocationCompute(value: string){
  computedLocation = // various google compute logic
  return computedLocation;
}

LocationInput(dsSource)
  const computeFunction = switch(dsSource)
      case "USGS_WATER"
            return googleLocationCompute
      case "waterSupply"
            return customCSVLocationCompute
       default:
           return locationCompute
}
<input onChange={(e) => computeFunction(e.value)} />
```

Now, I know google's compute function needs a useEffect, so it needs to be in a real component. This way wouldn't work but it's what I would've done to write the cleanest code. 
The goal here is to keep the input UI consistent but make sure the location change logic is different. So we can do pass the input component as a child, then have different components for each different dsSource. ex (https://www.dhiwise.com/post/react-pass-component-as-prop-guide-to-component-composition)

```
LocationSearchBar
   const renderSearchBar = switch (dsSource)
       case "USGS_WATER"
           <GoogleSearchBar inputComponent={DebouncedInput} />
        case "DEQ_WaterSuppy"
            <WaterSupplySearchBar inputComponent={DeboucedInput} />
        default:
              <DefaultLocationSearchBar inputComponent={DebouncedInput} />
  return {renderSearchBar()}
```